buildSrc/src/main/groovy/io/deephaven/project/ProjectType.groovy

```diff
@@ -0,0 +1,88 @@
+package io.deephaven.project
+
+import groovy.transform.CompileStatic
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.tasks.TaskProvider
+
+@CompileStatic
+class ProjectType {
+
+    public static final String VERIFY_ALL_PROJECTS_REGISTERED_TASK_NAME = 'verifyAllProjectsRegistered'
+
+    public static final String VERIFY_RUNTIME_CLASSPATH_IS_PUBLIC_TASK_NAME = 'verifyRuntimeClasspathIsPublic'
+
+    static void registerRoot(Project project) {
```

Comment:
Can we have more than one root?

Reply:
Technically yes, some other project could register itself as a "root" project. I *could* provide some safety rails, but I think it's sort of a harmless misconfiguration that should be immediately obvious.